src/store/persistence.js

```diff
@@ -0,0 +1,96 @@
+import VuexPersistence from "vuex-persist";
+import localForage from "localforage";
+import debounce from "lodash.debounce";
+
+// A subclass for VuexPersintence to deal with single namepaced modules and large collections
+// Should be provided with a module and an async storage
+// IMPORTANT: This class assumes a lot about the structure of state and mutations available in each module
+// If this structure changes, we inevitably have to update this class
+class VuexPersistentModule extends VuexPersistence {
+  constructor(module, storage) {
+    // Write startLoading and the main collection (as an array) to storage
+    async function saveModule(module, state, storage) {
+      const modifiedState = {
+        startLoading: state[module].startLoading,
+      };
+      modifiedState[module] = Object.values(state[module][module]);
+      await storage.setItem(module, modifiedState);
+    }
+
+    // Set all options in parent class
+    super({
+      storage,
+      asyncStorage: true,
+      modules: [module],
+      key: module,
+      filter: (m) => m.type.substring(0, m.type.indexOf("/")) === module,
+      saveState: debounce(saveModule, 60000),
+    });
+
+    // Get item from an async storage and commit them back to the store
+    this.replaceState = async (module, storage, store) => {
+      const savedState = await storage.getItem(module);
+      if (savedState) {
+        if (savedState.startLoading) {
+          store.commit(`${module}/setStartLoading`, savedState.startLoading);
+        }
+        if (savedState[module]) {
+          store.commit(
+            `${module}/set${module.charAt(0).toUpperCase() + module.slice(1)}`,
```

Comment:
Don't really like the manual capitalize. Can we not pass this in the constructor?

Reply:
I'm not sure I follow. Do you mean to to have a separate argument with the module name capitalized? That seems redundant to me (or we would just do this function inside the map where we create the instances).
Two other options would be:
* To extract this to a separate function, so we can just call `capitalize(module)`
* Pass the constructor a capitalized module name and lowercase it for all other places where we need it

Reply:
I'd prefer the second option then.